packages/webapp/pages/devcard.tsx

```diff
@@ -150,7 +150,13 @@ const Step2 = ({ initialDevCardSrc }: Step2Props): ReactElement => {
     () => generateQueryKey(RequestKey.DevCard, { id: user?.id }),
     [user],
   );
-  const devcard = client.getQueryData<GenerateDevCardParams>(key);
+  const [devcard, setDevCard] = useState(
+    client.getQueryData<GenerateDevCardParams>(key),
+  );
+  useEffect(() => {
```

Comment:
https://dailydotdev.slack.com/archives/C02E2C3C13R/p1707919572326829

Reply:
BTW we have `usePersistentContext` would that solve this maybe?
It's just a wrapper

Reply:
Why do we even use state here, we can just always update query cache from mutation and use that as state with `useQuery`?

Reply:
Well, I didn't say it's a good fix 😅

Reply:
So @kopancek planning on aligning with feedback, or we want to merge this as is to fix the bug for the moment? Not sure we're clear on the action?

Reply:
Working on a simplified version now :)

Reply:
What I mean the whole `setUpdatePreference` and its `useState` can just always be pulled from devcard right, its the same params? And then instead of setting state you would always mutate query cache. The initial data can be added from `useQuery. initialData`. And then you would not need the `onSuccess` since it would happen as soon the devcard is fetched for the first time and exists.

Reply:
Hmm, that's true. Except the `type` param, which is not part of the query, but we can keep that as a separate state. I can rewrite again with this in mind 👍

Reply:
@capJavert should be changed now accordingly 🙇